### PR TITLE
Add BBcodeBundle

### PR DESCRIPTION
**Pull request by itkg-ngilain**

### CODE REVIEW

File: BBcodeBundle/Definition/BBcodeDefinitionInterface.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
+<?php 
+
+namespace OpenOrchestra\BBcodeBundle\Definition;
+
+/**
+ * Interface BBcodeDefinition
```
Comment (itkg-alavieille) [inline BBcodeBundle/Definition/BBcodeDefinitionInterface.php]: BBcodeDefinitionInterface

File: BBcodeBundle/OpenOrchestraBBcodeBundle.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace OpenOrchestra\BBcodeBundle;
+
+use OpenOrchestra\BBcodeBundle\DependencyInjection\Compiler\BBcodeCompilerPass;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\HttpKernel\Bundle\Bundle;
+
+class OpenOrchestraBBcodeBundle extends Bundle
```
Comment (itkg-alavieille) [inline BBcodeBundle/OpenOrchestraBBcodeBundle.php]: comment class

File: BBcodeBundle/Parser/BBcodeParser.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace OpenOrchestra\BBcodeBundle\Parser;
+
+use JBBCode\Parser;
+use JBBCode\CodeDefinitionSet;
+use JBBCode\InputValidator;
+
+/**
+ * Class Parser
```
Comment (itkg-alavieille) [inline BBcodeBundle/Parser/BBcodeParser.php]: BBcodeParser

File: BBcodeBundle/Parser/BBcodeParser.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+ */
+class BBcodeParser
+{
+    protected $parser;
+
+    /**
+     * Constructor
+     */
+    public function __construct(Parser $parser, CodeDefinitionSet $codeDefinitionSet)
```
Comment (itkg-alavieille) [inline BBcodeBundle/Parser/BBcodeParser.php]: parameters in comment

**Comment by itkg-ngilain** [comment]
@itkg-alavieille : update done

File: BBcodeBundle/DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getConfigTreeBuilder()
+    {
+        $treeBuilder = new TreeBuilder();
+        $rootNode = $treeBuilder->root('open_orchestra_bbcode');
+
+        // Here you should define the parameters that are allowed to
```
Comment (itkg-alavieille) [inline BBcodeBundle/DependencyInjection/Configuration.php]: can you add configuration of extensions and definitions

**Comment by itkg-ngilain** [comment]
@itkg-alavieille , @itkg-nanne : pr updated

**Comment by itkg-ngilain** [comment]
@itkg-alavieille , @itkg-nanne : pr updated

File: BBcodeBundle/DependencyInjection/Compiler/BBcodeCompilerPass.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+     *
+     * @api
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $this->loadElements('validators', $container, 'loadValidatorsFromConfiguration', 'loadValidatorsFromService');
+        $this->loadElements('code_definitions', $container, 'loadDefinitionsFromConfiguration', 'loadDefinitionsFromService');
+    }
+
+    protected function loadElements($tagName, ContainerBuilder $container, $methodForConfiguration, $methodForService)
```
Comment (itkg-alavieille) [inline BBcodeBundle/DependencyInjection/Compiler/BBcodeCompilerPass.php]: method comment

File: BBcodeBundle/Tests/Parser/BBcodeParserTest.php [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    /**
+     * Set up the test
+     */
+    public function setUp()
+    {
+        $this->jparser = Phake::mock('JBBCode\Parser');
+        $this->parser = new BBcodeParser($this->jparser);
+    }
+
+    public function testGetAsHtml()
```
Comment (itkg-alavieille) [inline BBcodeBundle/Tests/Parser/BBcodeParserTest.php]: method comment

File: BBcodeBundle/Tests/Parser/BBcodeParserTest.php [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        $text = 'Some Random Text';
+        $this->parser->getAsHtml($text);
+
+        Phake::verify($this->jparser)->parse($text);
+        Phake::verify($this->jparser)->getAsHTML();
+    }
+
+    public function testGetAsBBCode()
```
Comment (itkg-alavieille) [inline BBcodeBundle/Tests/Parser/BBcodeParserTest.php]: method comment

File: BBcodeBundle/Tests/Parser/BBcodeParserTest.php [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    {
+        $text = 'Some Random Text';
+        $this->parser->getAsBBcode($text);
+
+        Phake::verify($this->jparser)->parse($text);
+        Phake::verify($this->jparser)->getAsBBCode();
+    }
+
+    public function testGetAsText()
```
Comment (itkg-alavieille) [inline BBcodeBundle/Tests/Parser/BBcodeParserTest.php]: method comment

**Comment by itkg-alavieille** [comment]
else methods comment :+1:

**Pull request merged**